src/pages/Admin/Charity/Permissions/Table.tsx

```diff
@@ -0,0 +1,129 @@
+import { useFormContext } from "react-hook-form";
+import { useAdminResources } from "pages/Admin/Guard";
+import { useGetWallet } from "contexts/WalletContext";
+import TableSection, { Cells } from "components/TableSection";
+import { CheckField } from "components/form";
+import { getTypedKeys } from "helpers";
+import LockButton from "./LockButton";
+import { FormField, FormValues, UpdateableFormValues } from "./schema";
+
+const formValues: UpdateableFormValues = {
+  accountFees: {} as FormField,
+  beneficiaries_allowlist: {} as FormField,
+  contributors_allowlist: {} as FormField,
+  donationSplitParams: {} as FormField,
+  profile: {} as FormField,
+};
+const FORM_KEYS = getTypedKeys(formValues);
+
+export default function Table({ className = "", disabled = false }) {
+  const { endow_type } = useAdminResources<"charity">();
+  const { wallet } = useGetWallet();
+
+  const {
+    register,
+    watch,
+    formState: { errors },
+  } = useFormContext<FormValues>();
+
+  const isNormal = endow_type === "normal";
+
+  return (
+    <table
+      className={`${className} w-full text-sm rounded border border-separate border-spacing-0 border-prim`}
+    >
+      <TableSection
+        type="thead"
+        rowClass="bg-orange-l6 dark:bg-blue-d7 divide-x divide-prim"
+      >
+        <Cells
+          type="th"
+          cellClass="py-3 px-4 text-xs uppercase font-semibold text-left first:rounded-tl last:rounded-tr"
+        >
+          <>Action</>
+          <>Admin wallet</>
+          {isNormal ? <>Governance</> : null}
+          <>Delegate</>
+          <>Delegate address</>
+          <>Actions</>
+        </Cells>
+      </TableSection>
+      <TableSection
+        type="tbody"
+        rowClass="dark:bg-blue-d6 divide-x divide-prim"
+        selectedClass="bg-orange-l5 dark:bg-blue-d4"
+      >
+        {FORM_KEYS.map((fieldName) => {
+          const delegate = watch(`${fieldName}.delegate`);
+          const name = watch(`${fieldName}.name`);
+          const initDelegate = watch(`initialValues.${fieldName}.delegate`);
+          const initDelegateAddress = watch(
+            `initialValues.${fieldName}.delegate_address`
+          );
+          const initModifiable = watch(`initialValues.${fieldName}.modifiable`);
+          const modifiable = watch(`${fieldName}.modifiable`);
+
+          const isDisabled =
+            disabled ||
+            !initModifiable ||
+            (initDelegate &&
+              !!initDelegateAddress &&
+              initDelegateAddress !== wallet?.address);
```

Comment:
@misicnenad,
[previous conversation](https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/1885#discussion_r1136490308) has been resolved on the basis of this new information ( regarding - `modifiable` field) 


>  We'll want to make sure that if a field is originally locked on contract then it can NOT be toggled back to unlocked. Allowing that would mean a submitted message / proposal would fail since it'd be trying to update a non-modifiable field in the EndowmentController.


and is partly covered in this check
```ts
const isDisabled =
            disabled ||
            !initModifiable ||
            (initDelegate &&
              !!initDelegateAddress &&
              initDelegateAddress !== wallet?.address);
```

but disabling the field based on who has current control delegated is still a mistake 

going back to this example (but without the `modifiable` part)

> let's say A, B, C are members of AdminX. and A creates a proposal to delegate all inividual settings to A ~~and set all fields to not modifiable~~

 ```
 //PERMISSIONS 
 1.`fees` --> delegate --> `A`
 2.`beneficiaries` --> delegate --> `A`
 3. `contributors` --> delegate --> `A`
```

this would mean that `B`,`C` can't propose to change the delegated address anymore even though it's `not locked`


NOTE: if `EndowmentController.endow_controller` row field is delegated to `A` then in this context (editing endowment controller as a `whole`) it makes sense to gray out this particular row - but it's not present atm in the UI 

but for other fields, checks should be done when editing those fields per [this comment](https://github.com/AngelProtocolFinance/angelprotocol-web-app/issues/1708#issuecomment-1469557854)


> This form is about updating the EndowmentController as a whole, and not the individual fields. The contract/delegate controllers of all of the other individual fields in that EndowmentController struct are only applicable when we are updating those fields themselves

Reply:
> So to determine whether a user is allowed to modify a specific row or not,

`B` and `C` should be able to modify every row (except `Endowment.endow_controller` row if it is included in this page ), since this particular page is about modifiying `EndowmentController` as a whole

the effects of other row/permission is manifested on pages of their scope 

Let's say after interacting with this page the ff. is the new permissions
```
controller (manifests in Edit Controller page) --> owner_controlled only
profile (manifests in Edit Profile page) -->  deligated only: to `A`
fees -(manifests in Fees page) -> deligated only:  to `A` 
```

it means if user `B` is trying to edit `EndowmentController` in edit controller page, he should be able to do so (and change every entries on it) since it is owner controlled 

but if user `B` tries to edit the `Profile` in edit profile page, he should not be able to do so since permission is only for `A`

..so forth for other pages scoped by a specific permission

Reply:
```
const userAuthorized: boolean = 
  (initDelegate && !!initDelegateAddress && initDelegateAddress !== wallet?.address) || (initOwnerControlled && propMeta.isAuthorized); // 1. or 2.
```
this logic is correct, given that `initDelegate` pertains only to `EndowController.endow_controller.delegate` since this particular permission is the one that matters to submit this edit controller page

Reply:
> ```
> const userAuthorized: boolean = 
>   (initDelegate && !!initDelegateAddress && initDelegateAddress !== wallet?.address) || (initOwnerControlled && propMeta.isAuthorized); // 1. or 2.
> ```
> 
> this logic is correct, given that `initDelegate` pertains only to `EndowController.endow_controller.delegate` since this particular permission is the one that matters to submit this edit controller page

Will update the code accordingly